src/runtime/TemporalObject.cpp

```diff
@@ -510,52 +663,345 @@ Value TemporalPlainDate::createTemporalDate(ExecutionState& state, const Value&
+            Value instant = TemporalInstant::createTemporalInstant(state, tItem->asTemporalZonedDateTimeObject()->getNanoseconds());
+            Value plainDateTime = TemporalTimeZone::builtinTimeZoneGetPlainDateTimeFor(state, tItem->asTemporalZonedDateTimeObject()->getTimeZone(), instant, tItem->asTemporalZonedDateTimeObject()->getCalendar());
+            return TemporalPlainDate::createTemporalDate(state, plainDateTime.asObject()->asTemporalPlainDateTimeObject()->getYear(), plainDateTime.asObject()->asTemporalPlainDateTimeObject()->getMonth(), plainDateTime.asObject()->asTemporalPlainDateTimeObject()->getDay(), plainDateTime.asObject()->asTemporalPlainDateTimeObject()->getCalendar());
+        } else if (tItem->isTemporalPlainDateTimeObject()) {
+            TemporalPlainDateTime* tmp = tItem->asTemporalPlainDateTimeObject();
+            return TemporalPlainDate::createTemporalDate(state, tmp->getYear(), tmp->getMonth(), tmp->getDay(), Value("iso8601"), new Object(state));
+        }
+        Value calendar = TemporalCalendar::getTemporalCalendarWithISODefault(state, item);
+        ValueVector requiredFields{
+            Value(new ASCIIString("day")),
+            Value(new ASCIIString("month")),
+            Value(new ASCIIString("monthCode")),
+            Value(new ASCIIString("year"))
+        };
+        ValueVector fieldNames = TemporalCalendar::calendarFields(state, calendar, ValueVector());
+        Value fields = Temporal::prepareTemporalFields(state, item, fieldNames, ValueVector());
+        return TemporalCalendar::dateFromFields(state, calendar, fields, options.value());
+    }
+    Temporal::toTemporalOverflow(state, options.value());
+    String* string = item.toString(state);
+    auto result = TemporalObject::parseTemporalDateString(state, string->toNonGCUTF8StringData());
+    ASSERT(TemporalPlainDate::isValidISODate(state, Value(result.at("year").c_str()).asInt32(), Value(result.at("month").c_str()).asInt32(), Value(result.at("day").c_str()).asInt32()));
+
+    Value calendar = TemporalCalendar::toTemporalCalendarWithISODefault(state, Value(result.at("calendar").c_str()));
+
+    return TemporalPlainDate::createTemporalDate(state, Value(result.at("year").c_str()).asInt32(), Value(result.at("month").c_str()).asInt32(), Value(result.at("day").c_str()).asInt32(), Value(result.at("calendar").c_str()), new Object(state));
+}
+
+std::map<std::string, int> TemporalPlainDate::balanceISODate(ExecutionState& state, int year, int month, int day)
+{
+    auto balancedYearMonth = TemporalPlainYearMonth::balanceISOYearMonth(state, year, month);
+    month = balancedYearMonth["Month"];
+    year = balancedYearMonth["Year"];
+    int testYear = year;
+
+    if (month <= 2) {
+        testYear--;
+    }
+
+    while (day < (-1 * TemporalCalendar::ISODaysInYear(state, testYear).asInt32())) {
+        day += TemporalCalendar::ISODaysInYear(state, testYear).asInt32();
+        year--;
+        testYear--;
+    }
```

Comment:
Are these loops complete in reasonable time? Is this come from the standard?

Reply:
Yes it did, but as i checked it they modified it and now it has 0 loops.